packages/keyring-controller/src/KeyringController.ts

```diff
@@ -823,13 +850,22 @@ export class KeyringController extends BaseController<
    * Gets the seed phrase of the HD keyring.
    *
    * @param password - Password of the keyring.
+   * @param keyringId - The id of the keyring.
    * @returns Promise resolving to the seed phrase.
    */
-  async exportSeedPhrase(password: string): Promise<Uint8Array> {
+  async exportSeedPhrase(
+    password: string,
+    keyringId?: string,
+  ): Promise<Uint8Array> {
     this.#assertIsUnlocked();
     await this.verifyPassword(password);
-    assertHasUint8ArrayMnemonic(this.#keyrings[0]);
-    return this.#keyrings[0].mnemonic;
+    const selectedKeyring = this.#getKeyringByIdOrDefault(keyringId);
```

Comment:
Defaulting to the primary HD keyring could give results that are unexpected for the caller. If the caller tries to export the seedphrase from an ID that for some reason is invalid it will receive the primary keyring seed.

Perhaps it would be more correct to let the operation fail.